src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -9,11 +9,13 @@
 
 public class ValuesResolver {
 
-
     public Map<String, Object> getVariableValues(GraphQLSchema schema, List<VariableDefinition> variableDefinitions, Map<String, Object> inputs) {
         Map<String, Object> result = new LinkedHashMap<String, Object>();
         for (VariableDefinition variableDefinition : variableDefinitions) {
-            result.put(variableDefinition.getName(), getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName())));
+            Object variableValue = getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName()));
+            if (variableValue != null) {
+                result.put(variableDefinition.getName(), variableValue);
+            }
```

Comment:
Sorry for the lack of explanation here. Indeed in the situation you mentiond here it didn't matter. But in cases the map is converted using jackson the key being absent is different than null value. One of the cases will trigger an unknown field exception.